memfs.go

```diff
@@ -381,10 +381,12 @@ func (d *fhDir) ReadDir(n int) ([]fs.DirEntry, error) {
 	return out, err
 }
 
+// File should not be shared among multiple Opens.
 type File struct {
 	name    string
 	perm    os.FileMode
 	content *bytes.Buffer
+	reader  io.ReadSeeker
```

Comment:
I don't see the point in having both `content` and `reader` as separate fields here. I think we can just make content be a *bytes.Reader. (It should be `*bytes.Reader`, not `io.ReadSeeker`, its a private field that is not polymorphic, we don't need to pretend like it is).

Reply:
agree, though i tried it first and and turns up not so wieldy since we need to explicitly save the file size for 
Stat. will try to make it work again though.

Reply:
You can use Seek to get the filesize

Reply:
ok, changed to the new approach which unfortunately requires a lock around the reader. lmk if that makes sense to you. imho i think the first approach is simpler.

Reply:
friendly ping